main.py

```diff
@@ -1,50 +1,175 @@
+"""
+    Usage: python3 main.py [-h] [-w] -u <homepage> [-p <project folder>] [-j <number of threads>]
+
+    Examples:
+        python3 main.py -p ../thenewboston -u https://thenewboston.com      # Specified project folder
+        python3 main.py -u https://thenewboston.com                         # Creates project folder thenewboston.com
+        python3 main.py -p thenewboston -u https://thenewboston.com -j20    # 20 threads
+        python3 main.py -h                                                  # Displays usage
+
+"""
+
 import threading
 from queue import Queue
 from spider import Spider
 from domain import *
 from general import *
+import getopt, sys
+import signal
 
-PROJECT_NAME = 'thenewboston'
-HOMEPAGE = 'https://thenewboston.com/'
-DOMAIN_NAME = get_domain_name(HOMEPAGE)
-QUEUE_FILE = PROJECT_NAME + '/queue.txt'
-CRAWLED_FILE = PROJECT_NAME + '/crawled.txt'
+
+PROJECT_NAME = ''
+HOMEPAGE = ''
+DOMAIN_NAME = ''
+QUEUE_FILE = ''
+CRAWLED_FILE = ''
 NUMBER_OF_THREADS = 8
 queue = Queue()
-Spider(PROJECT_NAME, HOMEPAGE, DOMAIN_NAME)
+
 
 
 # Create worker threads (will die when main exits)
 def create_workers():
-    for _ in range(NUMBER_OF_THREADS):
-        t = threading.Thread(target=work)
-        t.daemon = True
-        t.start()
+	for _ in range(NUMBER_OF_THREADS):
+		t = threading.Thread(target=work)
+		t.daemon = True
+		t.start()
 
 
 # Do the next job in the queue
 def work():
-    while True:
-        url = queue.get()
-        Spider.crawl_page(threading.current_thread().name, url)
-        queue.task_done()
+	while True:
+		url = queue.get()
+		Spider.crawl_page(threading.current_thread().name, url)
+		queue.task_done()
 
 
 # Each queued link is a new job
 def create_jobs():
-    for link in file_to_set(QUEUE_FILE):
-        queue.put(link)
-    queue.join()
-    crawl()
+	for link in file_to_set(QUEUE_FILE):
+		queue.put(link)
+		queue.join()
+		crawl()
 
 
 # Check if there are items in the queue, if so crawl them
 def crawl():
-    queued_links = file_to_set(QUEUE_FILE)
-    if len(queued_links) > 0:
-        print(str(len(queued_links)) + ' links in the queue')
-        create_jobs()
+	queued_links = file_to_set(QUEUE_FILE)
+	if len(queued_links) > 0:
+		print(str(len(queued_links)) + ' links in the queue')
+		create_jobs()
+
+
+# Cleanup and quit
+def quit_gracefully(signal=None, frame=None):
+	print("\nQuitting.")
+	with queue.mutex: queue.queue.clear()
+	del queue
```

Comment:
Maybe make use of the delete_file_contents() function in _general.py_ and cleanup the queue file?